### ISSUE TITLE
"Mark as unread" feature for notifications

### ISSUE THREAD

**Issue by patcon** [isaacs/github#236]
Ref: https://github.com/lorenzhs/hubot-tell/pull/11#issuecomment-46025737

cc: @lorenzhs

---

Changed issue title from _"Mark as unread feature for issues"_

**Comment by jmm** [comment]
+1

**Comment by brendensoares** [comment]
+1

**Comment by Wilkins** [comment]
:+1:

**Comment by tolry** [comment]
:+1:

**Comment by brianclements** [comment]
+many. I miss so many things because of this.

**Comment by bixycler** [comment]
:+1: Up up for this! I'm begging for an "unread" function.

**Comment by MattDiMu** [comment]
+1

**Comment by mike-charles** [comment]
:+1:

**Comment by bradt** [comment]
:+1: Again. Still need this.

**Comment by DavidBadura** [comment]
:+1:

**Comment by 743** [comment]
:+1:

**Comment by Reiner030** [comment]
(y)

**Comment by billwanjohi** [comment]
:+1:

**Comment by erkinalp** [comment]
@github: This feature may have an impact on some data-flow assumptions of the database.

**Comment by leilatoo** [comment]
:+1:

**Comment by mistakevin** [comment]
+1

**Comment by runofthemill** [comment]
+2

**Comment by JerryLopez** [comment]
+1

**Comment by roham** [comment]
I've been watching this thread for over a year, last week [ZenHub.io](https://zenhub.io) released a feature that makes the use-case super easy:
* [Introducing ToDo: the first personal productivity task manager built for GitHub](https://www.zenhub.io/blog/to-do-productivity-feature/)
* https://www.zenhub.io/blog/to-do-productivity-feature/

Disclosure: I'm a ZenHub cofounder and would love to hear your thoughts on the product!

**Comment by jansokoly** [comment]
Much needed. +1

**Comment by runofthemill** [comment]
@roham seems useful, but highly limited since ZenHub is paid and for teams. If the ToDo features were available for free and/or to individuals, then it might be a realistic solution to this issue. Regardless, ZenHub is a great product and kudos on making it even better :)

**Comment by jdfreder** [comment]
+1

**Comment by swapnilmmane** [comment]
+1 sounds good feature! :+1:

**Comment by s0undt3ch** [comment]
:+1: 

and ZenHub only solves part of the issue... I often check my notifications on my phone(where there's no ZenHub support), so, the mark as unread would be a major plus(but on the actual issue)

**Comment by 304NotModified** [comment]
+1

**Comment by roham** [comment]
[ZenHub](https://zenhub.io) is actually **completely 100% free** for personal, open-source, academic, and non-profit use. Not just ToDo but the entire PM stack is yours for free. 

ZenHub for mobile is coming as soon as we have enough demand for it.

**Comment by mgsloan** [comment]
+1 This actually seems possible [with the API](https://developer.github.com/v3/activity/notifications/#mark-as-read) (just set the timestamp backwards in time), so a chrome plugin could be written that has this feature.

**Comment by mfeliciano** [comment]
+1

**Comment by krodyrobi** [comment]
+1

**Comment by hdrdavies** [comment]
:+1:

**Comment by mfeliciano** [comment]
+1

**Comment by azachar** [comment]
+1 It is really hard to follow if you want to come back to some messages. You have no idea where it was if you are subscribed to many messages. I wonder why such a simple thing is missing :)

**Comment by mrmayfield** [comment]
+52

**Comment by ponelat** [comment]
+53

**Comment by zaininfo** [comment]
+∞

**Comment by DrewML** [comment]
> +1 This actually seems possible with the API (just set the timestamp backwards in time), so a chrome plugin could be written that has this feature.

For anyone wanting to try this out, I did a quick test with [that endpoint](https://developer.github.com/v3/activity/notifications/#mark-as-read), and unfortunately, GitHub's API won't let you trick it into marking something as unread :(

**Comment by etipton** [comment]
+1

**Comment by vinyar** [comment]
💯 💯

**Comment by mrmayfield** [comment]
Why is @TPS thumbing down everyone supporting this feature?

**Comment by zaininfo** [comment]
I guess he/she's a bit annoyed with the frequent bumping.
Frankly, this is a feature (marking something as unread) that most social platforms introduce after user's request, in spite of the fact that it's a basic utility feature.
Someone else might have suggested this too, but I have to use Gmail to track read/unread GitHub notifications, so much for the GitHub notifications page.

**Comment by TPS** [comment]
_(@zaininfo is correct, & https://github.com/isaacs/github/issues/640#issue-146807201 & https://github.com/isaacs/github/issues/640#issuecomment-213444485 elaborate._

_I apologize for my own spam, folks. :bow:)_

**Comment by jmm** [comment]
@mrmayfield @zaininfo **TPS** has also been spamming people on my issue, https://github.com/isaacs/github/issues/283, with :-1: for +1'ing it even though +1'ing actually relates to the nature of the issue and I encourage people to do it :/

(Aside: seriously, you can react to your own posts?)

**Comment by rhansen** [comment]
FYI, in [GitLab](https://gitlab.com/), notifications become "todo" items that you can check off (and uncheck later if needed).  It's very handy.

**Comment by sjackman** [comment]
:+1:

**Comment by guylepage3** [comment]
+1

**Comment by sganslandt** [comment]
👍 

This is I think the sole reason we're currently paying for ZenHub on top of GitHub.

**Comment by ericadamski** [comment]
👍

**Comment by pensierinmusica** [comment]
+1

**Comment by probonopd** [comment]
👍 
+10

**Comment by z3t0** [comment]
Just a note... It's a lot cleaner if people don't use the comments to vote. Please just 👍 the original comment. That way any thoughts or ideas about the actual request can be made in the comments... instead of essentially a host of votes that don't really add any new information

**Comment by larsborn** [comment]
we (@DavidBadura and I) got tired of waiting and hacked together a Chrome extension, that implements something along this lines: very similar to gitlab, one can manage a todo list within github. 

For this, the extension injects a button on every issue and pull request page with "Add ToDo". Also, it adds a button to the main toolbar at the top, that simply displays all ToDos in a grouped list. All data is stored in the browser's local storage so no data-hoarding on our side ;).

https://github.com/larsborn/GithubToDos 

We are open to feedback of course! Open an issue or drop us a message!

**Comment by z3t0** [comment]
@larsborn that looks really cool! Good luck. I'd help but I primarily use Safari :/

**Comment by ReikaKalseki** [comment]
*Angry dragon face*

For such a simple and normally ubiquitous feature, this is taking a long time to be implemented.

**Comment by juliangums** [comment]
+1

**Comment by bcx** [comment]
+1 

Being able to mark notifications as unread would be a big workflow help.

**Comment by r10s** [comment]
I would also like this feaure.
. when I got a notification I often have a quick look at it and decide to answer it later (maybe because of not time)
- these messages get easily missed later :(

**Comment by balazsdukai** [comment]
+1

**Comment by carlfredrikhero** [comment]
+1

**Comment by cnlohr** [comment]
Come on, guys... it's been three years and I need this button in my life.

**Comment by bfred-it** [comment]
[Refined GitHub](https://github.com/sindresorhus/refined-github) has exactly this feature :tada:

**Comment by probonopd** [comment]
> Refined GitHub has exactly this feature tada

How does it do that? Does GitHub offer an API to mark something as unread?

**Comment by bfred-it** [comment]
They're not saved back to GitHub but locally. Then it injects them into the Notifications page, blending them with the others notifications ([almost too well](https://github.com/sindresorhus/refined-github/issues/845))

**Comment by probonopd** [comment]
Uh. So it doesn't work across different machines? In that case, it won't help for glancing quickly into an issue on the mobile, and then setting it to "unread" in order to later work on it on the desktop.

**Comment by bfred-it** [comment]
No it wouldn't, unless you use Firefox mobile and RGH also works there and [the sync feature is added](https://github.com/sindresorhus/refined-github/issues/1036)

**Comment by cnlohr** [comment]
I don't understand how some of you guys are acting like using an add-on is a remotely acceptable solution to something that is lacking in core basic functionality.

**Comment by sami602** [comment]
+1

**Comment by probonopd** [comment]
I was greeted by this today. Is this the feature we all were after for so long?

![Screnshot](https://user-images.githubusercontent.com/2480569/47933653-0e0a0600-decd-11e8-88af-86f361079a3b.png)

**Comment by becca** [comment]
👋 we just shipped the ability to mark a notification as unread for web notifications (on desktop and mobile)!

**Comment by probonopd** [comment]
Thank you very much @becca. Now it would be very cool if we could mark as unread right from inside an issue, without having to go to the list of read issues first - mine is hundreds of entries long.

Think of the issues themselves as the items that are read or unread, not in terms of "notifications" but in the way an e-mail can be read or unread.

**Comment by bfred-it** [comment]
@becca how's that different from "Save for later" in practice? Looks like a duplicate feature

**Comment by becca** [comment]
@bfred-it great question. save for later was the first, quickest path to getting closer to mark as unread. i'm tracking usage of both and will likely suggest deprecating save for later as `mark as unread` was the customer ask that we needed to fulfill.

**Comment by etipton** [comment]
Any chance of making the bell icon in the upper-right become re-highlighted when going from 0 to 1 unread notifications?

**Comment by TPS** [comment]
[This is now mostly solved](https://github.com/isaacs/github/issues/236#issuecomment-455270131) from https://github.com/notifications/read : Click "•" on the far right of the issue & that converts back to the unread "✓".

This is opposed to #456, where I don't find any way to do this from w/*in* the issue/thread.

If any concerns, please comment!

**Comment by probonopd** [comment]
The way it is implemented right now is not very intuitive. It is way too much work to mark a ticket that you have just read as unread.

**Comment by m-kuhn** [comment]
Is there a possibility to mark it as unread from within the issue itself?

**Comment by Qix-** [comment]
@becca Agreed with others, it could use some love. I didn't realize this was even a thing until today - just being able to put _anything_ back into my notifications (a commit, an issue, a PR, etc.) to remind me to address it again later, even if it wasn't originally a notification, would be invaluable.

Most of the time my issues or PRs go unattended for so long (months/years) is due to me reading the notification on my phone and not being reminded to address it later when I'm at a computer.
